### PR TITLE
[SPARK-44994][PYTHON][DOCS] Refine docstring of DataFrame.filter

### PR DESCRIPTION
**Pull request by allisonwang-db**
### What changes were proposed in this pull request?

This PR refines the docstring of `DataFrame.filter` by adding more examples.

### Why are the changes needed?

To improve PySpark documentation.

### Does this PR introduce _any_ user-facing change?

No

### How was this patch tested?

doctest

### Was this patch authored or co-authored using generative AI tooling?

No

### CODE REVIEW

**Review by dongjoon-hyun — COMMENTED** [review]
Could you fix Python linter failure?
```
starting black test...
black checks failed:
would reformat python/pyspark/sql/dataframe.py

Oh no! 💥 💔 💥
1 file would be reformatted, 712 files would be left unchanged.
Please run 'dev/reformat-python' script.
1
Error: Process completed with exit code 1.
```

**Review by HyukjinKwon — APPROVED** [review]
LGTM (other than the linter failure)

File: python/pyspark/sql/dataframe.py [hunk 1]
```diff
@@ -3355,48 +3355,131 @@ def filter(self, condition: "ColumnOrName") -> "DataFrame":
...
+        >>> df.filter("age > 3 AND name = 'Bob'").show()
+        +---+----+-------+
+        |age|name|subject|
+        +---+----+-------+
+        |  5| Bob|Physics|
+        +---+----+-------+
+
+        Filter using the :func:`Column.isin` function.
+
+        >>> from pyspark.sql.functions import col
```
Comment (zhengruifeng) [inline python/pyspark/sql/dataframe.py:3431]: what about using
```
from pyspark.sql import functions as sf
df.filter(sf.col("name").isin("Alice", "Bob")).show()
```

to follow the example?

Comment (allisonwang-db) [reply python/pyspark/sql/dataframe.py:3431]: Sure!

**Comment by allisonwang-db** [comment]
cc @zhengruifeng @HyukjinKwon I've updated the PR. The test failures seem unrelated.

**Comment by zhengruifeng** [comment]
merged to master

**Pull request closed without merging**